src/control/control.c

```diff
@@ -225,6 +225,11 @@ void dt_control_draw_busy_msg(cairo_t *cr, int width, int height)
   pango_layout_set_font_description(layout, desc);
   pango_layout_set_text(layout, _("working..."), -1);
```

Comment:
Why not just:

```
pango_layout_set_font_description(layout, desc);

if(ink.width > width * 0.98)
   pango_layout_set_text(layout, "...", -1);
else
   pango_layout_set_text(layout, _("working..."), -1);

pango_layout_get_pixel_extents(layout, &ink, NULL);
```

Reply:
I'm a bigger fan of the ?: operator than most.

Reply:
Yes ?: works for me too.

Reply:
Ah but wait, here we first have to try to print the full text, to see how wide it would be and only then can we compare it to the thumbnail width...

Reply:
I see, the `pango_layout_get_pixel_extents(layout, &ink, NULL);` reset ink with new possible width !

Seems all good then... Anyway I'll test that tomorrow now.